src/test/java/com/synopsys/integration/detect/ApplicationUpdaterTest.java

```diff
@@ -1,44 +1,150 @@
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_DETECT_SOURCE)).thenReturn(fakeUrl);
+        Assertions.assertFalse(new ApplicationUpdater(mockedUtility, successArgs).canSelfUpdate());
+    }
+    
+    @Test
+    public void testCanSelfUpdateWithSystemEnvLatestRelease() {
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_DETECT_LATEST_RELEASE_VERSION)).thenReturn(fakeUrl);
+        Assertions.assertFalse(new ApplicationUpdater(mockedUtility, successArgs).canSelfUpdate());
+    }
+    
+    @Test
+    public void testCanSelfUpdateWithSystemEnvDetectVerKey() {
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_DETECT_VERSION_KEY)).thenReturn(fakeUrl);
+        Assertions.assertFalse(new ApplicationUpdater(mockedUtility, successArgs).canSelfUpdate());
+    }
+    
+    @Test
+    public void testNoProxyFoundOne() {
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_PROXY_HTTP_HOST)).thenReturn(fakeUrl);
+        Assertions.assertEquals(ProxyInfo.NO_PROXY_INFO, new ApplicationUpdater(mockedUtility, failureArgs).getProxyInfo());
+    }
+    
+    @Test
+    public void testNoProxyFoundTwo() {
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_PROXY_HTTP_PORT)).thenReturn(fakeUrl);
+        Assertions.assertEquals(ProxyInfo.NO_PROXY_INFO, new ApplicationUpdater(mockedUtility, failureArgs).getProxyInfo());
+    }
+    
+    @Test
+    public void testProxyFoundOne() {
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_PROXY_HTTP_HOST)).thenReturn(fakeUrl);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_PROXY_HTTP_PORT)).thenReturn("8080");
+        Assertions.assertNotEquals(ProxyInfo.NO_PROXY_INFO, new ApplicationUpdater(mockedUtility, failureArgs).getProxyInfo());
+    }
+    
+    @Test
+    public void testProxyFoundTwo() {
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_PROXY_HTTPS_HOST)).thenReturn(fakeUrl);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_PROXY_HTTPS_PORT)).thenReturn("8080");
+        Assertions.assertNotEquals(ProxyInfo.NO_PROXY_INFO, new ApplicationUpdater(mockedUtility, failureArgs).getProxyInfo());
```

Comment:
Proxy tests: If both proxy HOST and PORT is specified as HTTPS, we have a proxy!